lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -91,27 +91,19 @@ bool PagerandoBinning::runOnModule(Module &M) {
     auto I = Bins.lower_bound(FnSize);
     if (I == Bins.end()) {  // No bin with enough free space
       Bin = BinCount++;
-      FreeSpace = BinSize;
+      FreeSpace = BinSize - (FnSize % BinSize);
```

Comment:
Good catch! I would even say that this is a bug, because it lowers security (larger than necessary bins), while the condition below is an optimization to avoid unnecessary mappings.